src/services/YotiService.ts

```diff
@@ -211,7 +211,8 @@ export class YotiService {
 			this.logger.info("Received response from Yoti for create /sessions");
 			return data.session_id;
 		} catch (error: any) {
-			this.logger.error({ message: "An error occurred when creating Yoti session", yotiErrorMessage: error.message, yotiErrorCode: error.code, messageCode: MessageCodes.FAILED_CREATING_YOTI_SESSION });
+			const xRequestId = error.response ? error.response.headers.get("x-request-id") : undefined;
+			this.logger.error({ message: "An error occurred when creating Yoti session", yotiErrorMessage: error.message, yotiErrorCode: error.code, messageCode: MessageCodes.FAILED_CREATING_YOTI_SESSION, xRequestId });
```

Comment:
Do you think we can add a unit test for this?

Reply:
I tried, but it was hard to see how could i inject the headers to the Error object we are returning while rejecting the axiosMock call. I tried returning Response object, but it wasnt recognised as an error and failed detecting the error object. Also tried using the MockAdaptor which had a few examples of replying back with required responses and headers, but that didnt work too.

Reply:
I see. Because we're using `mockRejectedValueOnce` we don't actually have to pass in an error as such, it will reject the promise with whatever we give it so you can pass it an object that mirrors the axios error if you like -

```		
const responseHeaders = new Headers();
responseHeaders.append("x-request-id", "dummy-request-id");

axiosMock.post.mockRejectedValueOnce({
  message: "Failed to create session",
  response: { headers: responseHeaders },
});

```
It's definitely not a perfect match but it at least allows us to test that we are getting an appending the right header

Reply:
We could even put a `code` in that object to also test `yotiErrorCode`

Reply:
Done. Realised that my previous tries were failing cause of .get :(. Didnt troubleshoot it much then. Thanks.

Reply:
Ah right I see you've removed the `get` which is interesting. Was it not needed in the end? This still works when testing against the API, right?

Reply:
Replaced it with error.response.headers["x-request-id"] instead which works both both and tested it with Yoti url too and could get the requestId similar to one attached as evidence.